tests/src/gui/testqgsdoublevalidator.cpp

```diff
@@ -0,0 +1,138 @@
+/***************************************************************************
+    testqgsdoublespinbox.cpp
+     --------------------------------------
+    Date                 : June 2020
+    Copyright            : (C) 2020 Sebastien Peillet
+    Email                : sebastien dot peillet at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "qgstest.h"
+
+#include "qgsdoublevalidator.h"
+#include <QLineEdit>
+
+class TestQgsDoubleValidator: public QObject
+{
+    Q_OBJECT
+  private slots:
+    void initTestCase(); // will be called before the first testfunction is executed.
+    void cleanupTestCase(); // will be called after the last testfunction was executed.
+    void init(); // will be called before each testfunction is executed.
+    void cleanup(); // will be called after every testfunction.
+
+    void validate();
+    void toDouble();
+
+  private:
+
+};
+
+void TestQgsDoubleValidator::initTestCase()
+{
+
+}
+
+void TestQgsDoubleValidator::cleanupTestCase()
+{
+}
+
+void TestQgsDoubleValidator::init()
+{
+}
+
+void TestQgsDoubleValidator::cleanup()
+{
+}
+
+void TestQgsDoubleValidator::validate()
```

Comment:
this is fine but I would recommend to use _data driven testing here, it will be easier to read the test and add new cases: https://doc.qt.io/qt-5/qttestlib-tutorial2-example.html

Reply:
I didn't know the trick, thanks for the documentation. I made the changes